tests/test_config.py

```diff
@@ -119,10 +122,56 @@ def test_default_settings(root_window, monkeypatch):
     )
     assert config.pop("nick") == config.pop("username") == config.pop("realname")
     assert config == {
-        "extra_notifications": [],
         "host": "irc.libera.chat",
         "joined_channels": ["##learnpython"],
         "password": None,
         "port": 6697,
         "ssl": True,
+        "extra_notifications": [],
+        "join_leave_hiding": {"show_by_default": True, "exception_nicks": []},
     }
+
+
+def test_join_part_quit_messages(alice, bob, wait_until, monkeypatch):
+    bob.entry.insert("end", "/join #lol")
+    bob.on_enter_pressed()
+    wait_until(lambda: "The topic of #lol is:" in bob.text())
+
+    alice.entry.insert("end", "/join #lol")
+    alice.on_enter_pressed()
+    wait_until(lambda: "The topic of #lol is:" in alice.text())
+    alice.entry.insert("end", "/part #lol")
+    alice.on_enter_pressed()
+
+    wait_until(lambda: "Alice joined #lol." in bob.text())
+    wait_until(lambda: "Alice left #lol." in bob.text())
+
+
+def test_join_part_quit_messages_disabled(alice, bob, wait_until, monkeypatch):
+    bob.entry.insert("end", "/join #lol")
+    bob.on_enter_pressed()
+    wait_until(lambda: "The topic of #lol is:" in bob.text())
+
+    # Configure Bob to ignore Alice joining/quitting
+    def bob_config(transient_to, initial_config):
+        new_config = copy.deepcopy(initial_config)
+        new_config["join_leave_hiding"]["exception_nicks"].append("aLiCe")
+        return new_config
+
+    monkeypatch.setattr("mantaray.config.show_connection_settings_dialog", bob_config)
+    bob.get_server_views()[0].show_config_dialog()
+
+    alice.entry.insert("end", "/join #lol")
+    alice.on_enter_pressed()
+    wait_until(lambda: "The topic of #lol is:" in alice.text())
+    alice.entry.insert("end", "Hello Bob")
+    alice.on_enter_pressed()
+    alice.entry.insert("end", "/part #lol")
+    alice.on_enter_pressed()
+
+    wait_until(
+        lambda: "Hello Bob" in bob.text()
+        and "Alice" not in bob.get_current_view().userlist.get_nicks()
+    )
+    assert "Alice joined #lol." not in bob.text()
+    assert "Alice left #lol." not in bob.text()
```

Comment:
These are fragile, because they could pass just because the message is slightly different. 

```suggestion
    assert "joined" not in bob.text()
    assert "left" not in bob.text()
    assert "parted" not in bob.text()
    assert "quit" not in bob.text()
```